### PR TITLE
fix: allow passing `variant` prop to all checkboxes / radios through use...Group hook

### PR DESCRIPTION
**Pull request by unekinn**
## Summary

**useCheckboxGroup** and **useRadioGroup**: these hooks now also accept the prop `variant`, which sets the variant for all the checkboxes or radios in the group.

## Checks

- [x] I have read the [contribution guidelines](https://github.com/digdir/designsystemet/blob/main/CONTRIBUTING.md)
- [x] I have added a changeset (run `pnpm changeset` if relevant)

### CODE REVIEW

**Comment by mimarz** [comment]
Waiting for reviewing this after we finish web-first docs.

**Review by mimarz — COMMENTED** [review]
Looks fine to me, just need to fix the conflicts before merge.

**Comment by unekinn** [comment]
> Looks fine to me, just need to fix the conflicts before merge.

Should we do anything about the comments from Copilot? Some of it does make sense 🤔

**Comment by mimarz** [comment]
> > Looks fine to me, just need to fix the conflicts before merge.
> 
> Should we do anything about the comments from Copilot? Some of it does make sense 🤔

Yeah, we should... Question is do we omit or pick from the components props. This comment raises a potential bug edge-case: https://github.com/digdir/designsystemet/pull/4864#discussion_r3256814143

File: packages/react/src/utilities/hooks/use-radio-group/use-radio-group.test.tsx [hunk 1]
```diff
@@ -121,6 +121,20 @@ describe('RadioGroup', () => {
...
+    const radios = labels.map((label) => screen.getByLabelText(label));
+    const fields = radios
+      .map((radio) => radio.parentElement)
+      .filter((x) => x !== null);
+
+    expect(fields).toHaveLength(labels.length);
+    for (const field of fields) {
+      expect(field.nodeName).toBe('DS-FIELD');
+      expect(field).toHaveAttribute('data-variant', 'outline');
+    }
```
[the remark this answers is not in the document]
Comment (unekinn) [reply packages/react/src/utilities/hooks/use-radio-group/use-radio-group.test.tsx:136]: This copilot comment is incorrect, `fields` is correctly narrowed and TypeScript builds just fine.

File: packages/react/src/utilities/hooks/use-checkbox-group/use-checkbox-group.test.tsx [hunk 2]
```diff
@@ -127,6 +127,20 @@ describe('CheckboxGroup', () => {
...
+    const checkboxes = labels.map((label) => screen.getByLabelText(label));
+    const fields = checkboxes
+      .map((checkbox) => checkbox.parentElement)
+      .filter((x) => x !== null);
+
+    expect(fields).toHaveLength(labels.length);
+    for (const field of fields) {
+      expect(field.nodeName).toBe('DS-FIELD');
+      expect(field).toHaveAttribute('data-variant', 'outline');
+    }
```
[the remark this answers is not in the document]
Comment (unekinn) [reply packages/react/src/utilities/hooks/use-checkbox-group/use-checkbox-group.test.tsx:142]: This copilot comment is incorrect, `fields` is correctly narrowed and TypeScript builds just fine.

**Comment by unekinn** [comment]
> > Should we do anything about the comments from Copilot? Some of it does make sense 🤔
> 
> Yeah, we should... Question is do we omit or pick from the components props. This comment raises a potential bug edge-case: [#4864 (comment)](https://github.com/digdir/designsystemet/pull/4864#discussion_r3256814143)

Updated, @mimarz.

Thinking about it, only `variant` makes a lot of sense to add at the moment. You could argue that `data-color` and `data-size` should be included as well, but on the other hand they should probably be set on the fieldset instead?

At least it's easier to add more props later than to remove them 😄

File: packages/react/src/utilities/hooks/use-radio-group/use-radio-group.ts [hunk 3]
```diff
@@ -106,11 +111,18 @@ export function useRadioGroup({
...
      */
     getRadioProps: (propsOrValue: string | GetRadioProps) => {
-      const props =
+      let groupProps:
+        | Omit<RadioProps, 'aria-label' | 'aria-labelledby'>
+        | undefined;
+      if (props) {
+        const { onChange, error, ...rest } = props;
+        groupProps = rest;
+      }
```
[the remark this answers is not in the document]
Comment (unekinn) [reply packages/react/src/utilities/hooks/use-radio-group/use-radio-group.ts:120]: I think this is a premature optimization. Moving parts of the process out of the getter function makes the code less readable for very little gain

**Review by mimarz — APPROVED** [review]
LGTM 👍
